config.go

```diff
@@ -603,21 +604,24 @@ func loadConfig() (*config, []string, error) {
 	oldTestNets = append(oldTestNets, filepath.Join(cfg.DataDir, "testnet"))
 	cfg.DataDir = filepath.Join(cfg.DataDir, netName(activeNetParams))
 
-	// Append the network type to the log directory so it is "namespaced"
-	// per network in the same fashion as the data directory.
-	cfg.LogDir = cleanAndExpandPath(cfg.LogDir)
-	cfg.LogDir = filepath.Join(cfg.LogDir, netName(activeNetParams))
+	if !cfg.NoFileLogging {
+		// Append the network type to the log directory so it is "namespaced"
+		// per network in the same fashion as the data directory.
+		cfg.LogDir = cleanAndExpandPath(cfg.LogDir)
+		cfg.LogDir = filepath.Join(cfg.LogDir, netName(activeNetParams))
+		// Initialize log rotation.  After log rotation has been initialized, the
+		// logger variables may be used.
+		initLogRotator(filepath.Join(cfg.LogDir, defaultLogFilename))
+	} else {
+		logRotator = nil
+	}
 
 	// Special show command to list supported subsystems and exit.
 	if cfg.DebugLevel == "show" {
```

Comment:
perhaps move this above `if !cfg.NoFileLogging {`

Reply:
sure, will do